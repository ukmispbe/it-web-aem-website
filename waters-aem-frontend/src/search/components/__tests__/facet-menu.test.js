import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import FacetMenu from '../facet-menu';

describe('Feature: FacetMenu Component', () => {
    const props = {
        heading: 'Filter By',
        selectedValue: 'Application Note',
        previousIcon: 'PATH',
        filterTags: <></>,
        onClear: jest.fn()
    };

    describe('Scenario: Rendering', () => {
        describe('When content type has been specified', () => {
            it('Then it should match snapshot', () => {
                const json = renderer.create(<FacetMenu {...props} />);

                expect(json).toMatchSnapshot();
            });
        });
    });

    describe('Scenario: User Interaction', () => {
        describe('When the heading is clicked', () => {
            it('Then it should call the clear handler property', () => {
                const spyOnClear = spyOn(props, 'onClear');
                const wrapper = shallow(<FacetMenu {...props} />);
                const heading = wrapper.find('a');

                heading.simulate('click');
                
                expect(spyOnClear).toHaveBeenCalled();

                spyOnClear.mockRestore();
            });
        });
    });
});