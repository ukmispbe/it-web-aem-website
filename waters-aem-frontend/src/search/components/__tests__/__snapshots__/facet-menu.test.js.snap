// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: FacetMenu Component Scenario: Rendering When content type has been selected specified Then it should martch snapshot 1`] = `
<div
  className="facet-menu-container"
>
  <div
    className="facet-menu-container__heading"
  >
    <div
      className="heading--with-selected-value"
    >
      <div
        className="back-btn"
      >
        <a
          href="javascript:void(0)"
          onClick={[MockFunction]}
        >
          <div />
           
          Filter By
        </a>
      </div>
      <h3>
        Application Note
      </h3>
    </div>
  </div>
  <div
    className="facet-menu-container__body"
  />
</div>
`;
